extensions/usage-metrics-dashboard/app.R

```diff
@@ -746,10 +747,41 @@ server <- function(input, output, session) {
     paste0("Updated ", format(usage_last_updated(), fmt))
   })
 
+  # JavaScript for persisting search terms across table rerenders
+  table_js <- "
+  function(el, x) {
+    const tableId = el.id;
+
+    // Initialize global store for search terms if needed
+    if (!window.currentSessionSearches) {
+      window.currentSessionSearches = {};
```

Comment:
Rather than directly manipulating a window attribute you could use the `localStorage` API https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage

Reply:
I used `sessionStorage` :)

Reply:
Smart I'll take a look at the new implementation.